src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -212,12 +219,53 @@ func CreateAdmin(
 		if err != nil {
 			return nil, err
 		}
+		auth, err := authConfig()
+		if err != nil {
+			return nil, err
+		}
+		if auth != nil {
+			cfg, err = kafka.ConfigureSASL(cfg, auth)
+			if err != nil {
+				return nil, err
+			}
+		}
 		bs := brokers()
 		admin, err := sarama.NewClusterAdmin(bs, cfg)
 		return admin, wrapConnErr(err, bs)
 	}
 }
 
+func KafkaAuthConfig(
+	user, password, mechanism *string,
+) func() (*config.SCRAM, error) {
+	return func() (*config.SCRAM, error) {
+		if *user == "" && *password == "" {
+			return nil, nil
```

Comment:
This return statement could return error with the information that there is no user and password.

Reply:
That wouldn't be an error, though. It just means that the user didn't pass the credentials.

Reply:
Ok.